spec/castle_devise/integration/password_update_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+RSpec.describe "Password update", type: :request do
+  subject(:send_password_update) do
+    # First, we need to sign in so we can update the password in the next step
+    send_sign_in_request(email, password, request_token)
```

Comment:
Let's not perform http requests to log in the user.

1. in spec_helper:
```
  config.include Devise::Test::IntegrationHelpers
```

2. In a before block in this file: `sign_in(user)`